File: nvxapp.server.service/ClientServer-Service/GestionePresenze/TimeSheet_EngineService/Models/TimeSheet_EngineModel.cs
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_AnagraficaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_CausaliService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_GiustificativiService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_ResultService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_RichiestaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_GG_TimbraturaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Dip_RapportoLavoroService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_OrarioIntervalloHHService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_OrarioService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ProfiloOrarioGGService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ProfiloOrarioService.Models;
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.TimeSheet_EngineService.Models
{
    public class TimeSheet_CalculateModel
    {

        public int Year { get; set; }
        public int Month { get; set; } = 0;
        public List<string> SelectedUserId { get; set; } = new List<string>();
        public DateTime Dal { get; set; }
        public DateTime Al { get; set; }

        public Boolean Approva_Richieste_Timbrature { get; set; }
        public Boolean Approva_Richieste_Giustificativo { get; set; }
        public Boolean Genera_Timbrature_Mancanti { get; set; }
        public Boolean Genera_Giustificativo_Assenza { get; set; }



    }
    public class TimeSheet_CalculateInModel
    {
        public TimeSheet_CalculateModel TimeSheet_Calculate { get; set; } = new TimeSheet_CalculateModel();
    }
    public class TimeSheet_CalculateOutModel : ModelResult
    {
        public TimeSheet_CalculateModel TimeSheet_Calculate { get; set; } = new TimeSheet_CalculateModel();

        public TimeSheet_CalculateOutModel()
        {

        }
    }







    // Una riga orario del giorno, corrispondente a una riga Par_ProfiloOrarioGG.
    // ZOrder 1 = orario base (sempre presente), ZOrder > 1 = override condizionale.
    public class Dip_ProfiloOrario_DaySlot_GG
    {
        public int ZOrder { get; set; } // ordinamento: 1 = default, 2+ = override
        public int IdPar_Orario { get; set; }
    }
    // Un record per ogni combinazione dipendente × giorno con tutte le righe orario ordinate per ZOrder
    public class Dip_ProfiloOrario_DaySlot
    {
        public string IdAspNetUsers { get; set; } = string.Empty;
        public int IdDip_RapportoLavoro { get; set; }
        public DateTime Data { get; set; }
        public int IdPar_ProfiloOrario { get; set; }
        // righe orario del giorno ordinate per ZOrder (1=base, 2+=override)
        public List<Dip_ProfiloOrario_DaySlot_GG> Orari { get; set; } = new List<Dip_ProfiloOrario_DaySlot_GG>();
    }


    public class OrariSchema_4User_InModel
    {
        public List<string> UsersId { get; set; } = new List<string>();
        public DateTime Dal { get; set; }
        public DateTime Al { get; set; }
    }
    public class OrariSchema_4User_OutModel : ModelResult
    {
        public List<Dip_AnagraficaModel> Dip_Anagrafica { get; set; } = new List<Dip_AnagraficaModel>();
        public List<Dip_RapportoLavoroModel> Dip_RapportoLavoro { get; set; } = new List<Dip_RapportoLavoroModel>();
        public List<Dip_ProfiloOrario_DaySlot> DaySlots { get; set; } = new List<Dip_ProfiloOrario_DaySlot>();
        public List<Par_ProfiloOrarioModel> Par_ProfiloOrario { get; set; } = new List<Par_ProfiloOrarioModel>();
        public List<Par_ProfiloOrarioGGModel> Par_ProfiloOrarioGG { get; set; } = new List<Par_ProfiloOrarioGGModel>();
        public List<Par_OrarioModel> ParOrario { get; set; } = new List<Par_OrarioModel>();
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();
    }



    public class Dip_GG_AllData_InModel
    {
        public List<string> UsersId { get; set; } = new List<string>();
        public DateTime Dal { get; set; }
        public DateTime Al { get; set; }
    }
    public class Dip_GG_AllData_OutModel : ModelResult
    {
        public List<Dip_GG_TimbraturaModel> Dip_GG_Timbratura { get; set; } = new List<Dip_GG_TimbraturaModel>();
        public List<Dip_GG_CausaliModel> Dip_GG_Causali { get; set; } = new List<Dip_GG_CausaliModel>();
        public List<Dip_GG_GiustificativiModel> Dip_GG_Giustificativi { get; set; } = new List<Dip_GG_GiustificativiModel>();
        public List<Dip_GG_RichiestaModel> Dip_GG_Richiesta { get; set; } = new List<Dip_GG_RichiestaModel>();
        public List<Dip_GG_ResultModel> Dip_GG_Result { get; set; } = new List<Dip_GG_ResultModel>();
    }


    public class Timesheet_AllData_InModel
    {
        public List<string>? UsersId { get; set; } 
        public DateTime Dal { get; set; }
        public DateTime Al { get; set; }
    }

    public class Timesheet_AllData_OutModel : ModelResult
    {
        public Dip_GG_AllData_OutModel Dip_GG_AllData_OutModel { get; set; } = new Dip_GG_AllData_OutModel();
        public OrariSchema_4User_OutModel OrariSchema_4User_OutModel { get; set; } = new OrariSchema_4User_OutModel();
        
        
    }

}

File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/TimeSheet_EngineService/Models/time-sheet-engine-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { Dip_AnagraficaModel } from '../../Dip_Anagrafica/Models/dip-anagrafica-model';
import { Dip_RapportoLavoroModel } from '../../Dip_RapportoLavoro/Models/dip-rapporto-lavoro-model';
import { Dip_GG_CausaliModel } from '../../Dip_GG_Causali/Models/dip-gg-causali-model';
import { Dip_GG_GiustificativiModel } from '../../Dip_GG_Giustificativi/Models/dip-gg-giustificativi-model';
import { Dip_GG_RichiestaModel } from '../../Dip_GG_Richiesta/Models/dip-gg-richiesta-model';
import { Dip_GG_TimbraturaModel } from '../../Dip_GG_Timbratura/Models/dip-gg-timbratura-model';
import { Par_OrarioIntervalloHHModel } from '../../Par_OrarioIntervalloHH/Models/par-orario-intervallo-hh-model';
import { Par_OrarioModel } from '../../Par_Orario/Models/par-orario-model';
import { Par_ProfiloOrarioGGModel } from '../../Par_ProfiloOrarioGG/Models/par-profilo-orario-gg-model';
import { Par_ProfiloOrarioModel } from '../../Par_ProfiloOrario/Models/par-profilo-orario-model';

// ─── Calculate ────────────────────────────────────────────────────────────────

export class TimeSheet_CalculateModel {
  year: number;
  month: number;
  selectedUserId: string[] | null = null;
  dal: string;
  al: string;
  approva_Richieste_Timbrature: boolean;
  approva_Richieste_Giustificativo: boolean;
  genera_Timbrature_Mancanti: boolean;
}

export class TimeSheet_CalculateInModel {
  timeSheet_Calculate: TimeSheet_CalculateModel;
  constructor() {
    this.timeSheet_Calculate = new TimeSheet_CalculateModel();
  }
}

export class TimeSheet_CalculateOutModel extends ModelResult {
  timeSheet_Calculate: TimeSheet_CalculateModel;
}

// ─── DaySlot helpers ──────────────────────────────────────────────────────────

export class Dip_ProfiloOrario_DaySlot_GG {
  zOrder: number;
  idPar_Orario: number;
}

export class Dip_ProfiloOrario_DaySlot {
  idAspNetUsers: string;
  idDip_RapportoLavoro: number;
  data: string;               // ISO date string
  idPar_ProfiloOrario: number;
  orari: Dip_ProfiloOrario_DaySlot_GG[] = [];
}

// ─── Get_OrariSchema_4User ────────────────────────────────────────────────────

export class OrariSchema_4User_InModel {
  usersId: string[] = [];
  dal: string;
  al: string;
}

export class OrariSchema_4User_OutModel extends ModelResult {
  dip_Anagrafica: Dip_AnagraficaModel[] = [];
  dip_RapportoLavoro: Dip_RapportoLavoroModel[] = [];
  daySlots: Dip_ProfiloOrario_DaySlot[] = [];
  par_ProfiloOrario: Par_ProfiloOrarioModel[] = [];
  par_ProfiloOrarioGG: Par_ProfiloOrarioGGModel[] = [];
  parOrario: Par_OrarioModel[] = [];
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[] = [];
}

// ─── Dip_GG_AllData ───────────────────────────────────────────────────────────

export class Dip_GG_AllData_InModel {
  usersId: string[] = [];
  dal: string;
  al: string;
}

export class Dip_GG_AllData_OutModel extends ModelResult {
  dip_GG_Timbratura: Dip_GG_TimbraturaModel[] = [];
  dip_GG_Causali: Dip_GG_CausaliModel[] = [];
  dip_GG_Giustificativi: Dip_GG_GiustificativiModel[] = [];
  dip_GG_Richiesta: Dip_GG_RichiestaModel[] = [];
}

// ─── Get_Timesheet_AllData ────────────────────────────────────────────────────

export class Timesheet_AllData_InModel {
  usersId: string[] | null = null;
  dal: string;
  al: string;
}

export class Timesheet_AllData_OutModel extends ModelResult {
  dip_GG_AllData_OutModel: Dip_GG_AllData_OutModel = new Dip_GG_AllData_OutModel();
  orariSchema_4User_OutModel: OrariSchema_4User_OutModel = new OrariSchema_4User_OutModel();
}




